app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/metrics_capture.rb

```diff
@@ -0,0 +1,151 @@
+    raise _("Missing monitoring instance id") if metrics_endpoint.nil?
+
+    instance_id = metrics_endpoint.options["monitoring_instance_id"]
+
+    response = RestClient::Request.execute(
+      :method  => :post,
+      :url     => "https://#{ext_management_system.provider_region}.monitoring.cloud.ibm.com/api/data",
+      :headers => {
+        'Content-Type'  => 'application/json',
+        'Authorization' => "Bearer #{iam_access_token}",
+        'IBMInstanceID' => instance_id
+      },
+      :payload => JSON.generate(get_metrics_query(target.name))
+    )
+    data = JSON.parse(response.body)
+    dataset = consolidate_data(data["data"])
+
+    store_datapoints!(dataset[:timestamps], dataset[:cpu_usage_rate_average], "cpu_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:mem_usage_absolute_average], "mem_usage_absolute_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:net_usage_rate_average], "net_usage_rate_average", counter_values_by_mor[target.ems_ref])
+    store_datapoints!(dataset[:timestamps], dataset[:disk_usage_rate_average], "disk_usage_rate_average", counter_values_by_mor[target.ems_ref])
+
+    return counters_by_mor, counter_values_by_mor
+  rescue RestClient::ExceptionWithResponse => err
+    log_header = "[#{interval_name}] for: [#{target.class.name}], [#{target.id}], [#{target.name}]"
+    _log.error("#{log_header} Unhandled exception during perf data collection: [#{err}], class: [#{err.class}]")
+    _log.log_backtrace(err)
+    raise
+  end
+
+  def get_metrics_query(instance_name)
+    {
+      :last           => 144_00,
+      :sampling       => 600,
+      :filter         => "ibm_resource_name = '#{instance_name}'",
+      :metrics        => [
+        {
+          :id           => "ibm_is_instance_cpu_usage_percentage",
+          :aggregations => {
+            :time => "avg"
+          }
+        },
+        {
+          :id           => "ibm_is_instance_memory_usage_percentage",
+          :aggregations => {
+            :time => "avg"
+          }
+        },
+        {
+          :id           => "ibm_is_instance_network_in_bytes",
+          :aggregations => {
+            :time => "avg"
+          }
+        },
+        {
+          :id           => "ibm_is_instance_network_out_bytes",
+          :aggregations => {
+            :time => "avg"
+          }
+        },
+        {
+          :id           => "ibm_is_instance_volume_read_bytes",
+          :aggregations => {
+            :time => "avg"
+          }
+        },
+        {
+          :id           => "ibm_is_instance_volume_write_bytes",
+          :aggregations => {
+            :time => "avg"
+          }
+        }
+      ],
+      :dataSourceType => "host"
+    }
+  end
+
+  def iam_access_token
+    IBMCloudSdkCore::IAMTokenManager.new(:apikey => ext_management_system.authentication_key("default")).access_token
```

Comment:
```suggestion
    @iam_access_token ||= IBMCloudSdkCore::IAMTokenManager.new(:apikey => ext_management_system.authentication_key("default")).access_token
```

should this be memo-ized as shown above?

Reply:
Yes, thank you!